common/beerocks/tlvf/src/include/beerocks/tlvf/beerocks_header.h

```diff
@@ -39,10 +39,31 @@ class beerocks_header : public ClassList {
     }
     beerocks_message::eAction action()
     {
-        return actionhdr() ? actionhdr()->action() : beerocks_message::ACTION_NONE;
+        auto action_hdr = actionhdr();
+        if (!action_hdr) {
+            return beerocks_message::ACTION_NONE;
+        } else {
+            return action_hdr->action();
+        }
+    }
```

Comment:
This is exactly the same as the original... Does klocwork really barf on `?:` expressions? Or isn't it be able to recognise that `actionhdr()` always returns the same value? Hm, probably the latter... Would it help if we would add `__attribute__((pure))` to the actionhdr declaration?

Oh, and we probably need to make all those functions const as well. But that's unrelated.

Reply:
Yes, klocwork complainants whenever we try to access the return value of any method that returns a shared pointer unless we validate that it is not null

Reply:
But ` = actionhdr() ? ACTION_NONE : actionhd->action();` is checking exactly the same thing. Weird.

Reply:
No, because there is no guarantee that actionhdr() will return the same value in both cases. That's why I suggested to declare it as pure - if klocwork knows the pure attribute, it can use it to determine that in fact the same value will be returned in both cases.